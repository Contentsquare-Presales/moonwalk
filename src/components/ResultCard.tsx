import React from "react";
import styled from "styled-components/native";
import { useTheme } from "@react-navigation/native";

const Wrapper = styled.TouchableOpacity`
  padding: 20px;
  height: 100px;
  width: 100%;
  justify-content: center;
  border-bottom-width: 1px;
`;

const Title = styled.Text`
  font-weight: bold;
  margin-bottom: 5px;
`;

const Subtitle = styled.Text``;

const ResultCard = ({ data, showDetails }) => {
  const { name, net } = data;
  const { colors } = useTheme();

  return (
    <Wrapper
      onPress={() => showDetails(data)}
      style={{
        backgroundColor: colors.secondary,
        borderBottomColor: colors.uiAccent,
      }}
    >
      <Title style={{ color: colors.text }}>{name}</Title>
      <Subtitle style={{ color: colors.text }}>{net}</Subtitle>
    </Wrapper>
  );
};

export default ResultCard;
