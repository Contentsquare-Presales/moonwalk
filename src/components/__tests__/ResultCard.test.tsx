import React from "react";
import ResultCard from "../ResultCard";
import renderer from "react-test-renderer";

describe("ResultCard", () => {
  it("renders correctly", () => {
    const props = {
      data: {
        name: "$_TEST_NAME_$",
        net: "$_TEST_NET_DATE_$"
      },
      showDetails: jest.fn()
    };
    const tree = renderer.create(<ResultCard {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
