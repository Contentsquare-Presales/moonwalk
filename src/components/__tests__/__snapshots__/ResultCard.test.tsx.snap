// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultCard renders correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": undefined,
      "borderBottomColor": undefined,
      "borderBottomWidth": 1,
      "height": 100,
      "justifyContent": "center",
      "opacity": 1,
      "paddingBottom": 20,
      "paddingLeft": 20,
      "paddingRight": 20,
      "paddingTop": 20,
      "width": "100%",
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "fontWeight": "bold",
          "marginBottom": 5,
        },
        Object {
          "color": "rgb(28, 28, 30)",
        },
      ]
    }
  >
    $_TEST_NAME_$
  </Text>
  <Text
    style={
      Array [
        Object {},
        Object {
          "color": "rgb(28, 28, 30)",
        },
      ]
    }
  >
    $_TEST_NET_DATE_$
  </Text>
</View>
`;
